dist/web3-light.js

```diff
@@ -1429,22 +1429,26 @@ web3.version.api = version.version;
 web3.eth = {};
 
 /*jshint maxparams:4 */
-web3.eth.filter = function (fil, eventParams, options, formatter) {
+web3.eth.filter = function (fil, eventParams, options, callback) {
+
+    if (utils.isFunction(arguments[arguments.length - 1])) {
+        callback = arguments[arguments.length - 1];
+    }
 
     // if its event, treat it differently
     // TODO: simplify and remove
     if (fil._isEvent) {
-        return fil(eventParams, options);
+        return fil(eventParams, options, callback);
     }
 
     // output logs works for blockFilter and pendingTransaction filters?
-    return new Filter(fil, watches.eth(), formatter || formatters.outputLogFormatter);
+    return new Filter(fil, watches.eth(), callback, formatters.outputLogFormatter);
```

Comment:
Sure, but you probably want to get the filter and then use it later in the callback to stop it:

```js
var filter = web3.eth.filter({}, funciton(e, log){

   ...
   filter.stopWatching();
})
```